dbms/src/Core/Defines.h

```diff
@@ -162,3 +162,12 @@
 /// A macro for suppressing warnings about unused variables or function results.
 /// Useful for structured bindings which have no standard way to declare this.
 #define UNUSED(...) (void)(__VA_ARGS__)
+
+/// A macro for declaring global symbols weak. Useful when certain global function
+/// may be substituted by different implementation by another application using
+/// ClickHouse as a library (i.e. checkStackSize()).
+#if defined(__GNUC__)
```

Comment:
We use attribute weak without any macro and it works fine both for `clang` and `gcc`; on Linux, FreeBSD, MacOS; on x86_64, aarch64, ppc64...

Just proceed without macro or replace all other usages of `__attribute__((__weak__))` in code.

Reply:
```
~/clickhouse/source/dbms $ fgrep -r "((__weak__))" .
~/clickhouse/source/dbms $ 
```

I did not find any occurrences of __attribute__((__weak__) in DBMS source code, can you point me to any of them.

Regarding the macro: current version is based on the Y_WEAK macro from Arcadia, I believe that __GNUC__ macro works for both g++ and clang, and it was introduced for compatibility with MSVC. Is ClickHouse buildable under MSVC? If not, I'll remove macro and proceed with raw __attribute__((__weak__)).